### PR TITLE
feat(starfish): add bounds when fetching for span samples

### PR DESCRIPTION
**Pull request by DominikB2014**
In order to prevent weird samples way higher then the p95 causing the duration yAxis to change significantly. We want to maintain the "quadrants" of the original p95 graph.

In addition to this change, I also had to update the `useSpansQuery` to be dynamically typed. I need more properties from `useQuery`, specifically `isFetching` as `isLoading` only applies to inital data loads and not on manual refetches. This is also a better sol'n imo as we get access to all the `useQueryResult` properties.

### CODE REVIEW

File: static/app/views/starfish/queries/useSpanSamples.tsx [hunk 1]
```diff
@@ -52,6 +67,8 @@ export const useSpanSamples = (options: Options) => {
         `${url}?${qs.stringify({
           ...dateCondtions,
           ...{utc: location.query.utc},
+          upperBound: computeAxisMax([spanMetricsSeriesData?.[`p95(${SPAN_SELF_TIME})`]]),
+          lowerBound: 0,
```
Comment (DominikB2014) [inline static/app/views/starfish/queries/useSpanSamples.tsx:71]: Fyi @wmak I’m curious, why is the lowerBound auto set to p50 in the backend? In this PR I set my lowerBound to 0, as i’m fairly sure the yAxis on the chart always starts at 0. But if there's a reason to do p50 lmk!

Comment (wmak) [reply static/app/views/starfish/queries/useSpanSamples.tsx:71]: nope 0 makes sense, p50 is the lowest percentile we can do in the metrics dataset

Comment (DominikB2014) [reply static/app/views/starfish/queries/useSpanSamples.tsx]: fyi @wmak, updated this to use `lower`, `upper`, `firstBound` and `secondBound`

File: static/app/views/starfish/queries/useSpanSamples.tsx [hunk 2]
```diff
@@ -54,6 +71,10 @@ export const useSpanSamples = (options: Options) => {
         `${url}?${qs.stringify({
           ...dateCondtions,
           ...{utc: location.query.utc},
+          lowerBound: 0,
+          firstBound: (maxYValue * 1) / 3,
+          secondBound: (maxYValue * 2) / 3,
```
Comment (wmak) [inline static/app/views/starfish/queries/useSpanSamples.tsx:76]: nit: why not 
```suggestion
          firstBound: maxYValue * (1 / 3),
          secondBound: maxYValue * (2 / 3),
```

**Review by wmak — APPROVED** [review]

File: static/app/views/starfish/queries/useSpanSamples.tsx (hunk 2 above)
Comment (DominikB2014) [reply static/app/views/starfish/queries/useSpanSamples.tsx]: Yeah i like that better too 👍 .

So originally wrote this as
image
But prettier autoformatted it to 
```firstBound: (maxYValue * 1) / 3```

Prettier seems cool with 
```firstBound: maxYValue * (1 / 3), ```
as well, so i'll update it

**Pull request merged**